### DISCUSSION TITLE
Easier access to launcher configuration

### DISCUSSION THREAD

**Discussion by Knyffen** [runelite/runelite#17652] [Ideas]
**The suggestion**
Add a button inside RuneLite to open the launcher configuration.
I suggest either adding it next to the "Join Discord" and "Sign in to RuneLite" buttons, or inside the menu of the "Info Panel" plugin.

**Why it is necessary**
Certain RuneLite settings can be set using the launcher configuration as described [on the wiki](https://github.com/runelite/runelite/wiki/RuneLite-Launcher-Configuration). One of these is the scaling parameter, which is very important when people use HIDPI displays.
I had quite a hard time finding said wiki page since when searching Google for "runelite scaling" the results are about scaling the game itself and not the runelite client. Only when searching for "runelite scaling java" did I find the wiki. But for the sake of the argument, lets assume the people have found said wiki page.

Linux users are asked to pass the `--configure` argument to open the menu, which has been fine up until now. The future however points to people launching RuneLite using the Jagex Launcher [which won't get official Linux support](https://help.jagex.com/hc/en-gb/articles/13413514881937-Downloading-the-Jagex-Launcher-on-Linux).

The Jagex Launcher can be run using Wine, but running the launcher installer is tricky due to a Wine error and the recommended solution is therefore to install the launcher using Flatpak. But when using this process there isn't any documentation on how to point the Jagex Launcher to the RuneLite AppImage, and instead people install RuneLine along with the Jagex Launcher Flatpak.

The result of all this is that the actual RuneLite jar file is hidden away behind a lot of complexity. With my installation, I found that /var/lib/flatpak/runtime/com.jagex.Launcher.ThirdParty.RuneLite/x86_64/stable/85068d51323554dbb930263b432b40fec3f558d53a6660f540af088d44452672/files/RuneLite.exe is actually a bash script that runs RuneLite.
I can therefore add the `--configure` flag there, but I wouldn't expect the average user to find that, why a button within RuneLite itself would be a huge improvement.

**Comment by daniyl-x** [comment]
I can confirm that this is relevant for me on Linux since it's basically required to use a Jagex launcher for auth and the official one isn't provided for Linux.

I tried to copy a settings.json file to the flatpak's home directory but it doesn't work for scaling parameter for some reason.

**Reply by cdfisher** [reply]
If you use Bolt, which I'd recommend for basically anyone playing on Linux, it's super easy to access the configuration through that.

**Reply by daniyl-x** [reply]
@cdfisher thank you, I saw it allows passing CLI arguments, but I would like to use the official launcher even if it requires to rely on compatibility layer or hacks to use it.

I saw some users had issues with Bolt, even if it was a small amount of them, so I don't really want to risk getting banned. I understand that Bolt is most likely totally safe and the dev team behind it is doing whatever it is in their effort to provide the best second option for Linux users, but account moderation still depends on Jagex.

It's really a shame they won't provide a way to reliably play for Linux users, who want to use an alternative client. Maybe some day this will change.
